### PR TITLE
BUG: sparse: stop comparing None and int in bsr_matrix constructor

### PR DESCRIPTION
**Pull request by ev-br**
Fixes gh-6292.

### CODE REVIEW

File: scipy/sparse/bsr.py [hunk 1]
```diff
@@ -165,6 +165,8 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False, blocksize=None):
                 maxval = None
                 if shape is not None:
                     maxval = max(shape)
+                else:
```
Comment (matthew-brett) [inline scipy/sparse/bsr.py]: Maybe `maxval = 1 if shape is None else max(shape)`?

Comment (perimosocordiae) [inline scipy/sparse/bsr.py]: Or just initialize `maxval = 1` on line 165 and leave the rest as it was.

File: scipy/sparse/tests/test_base.py [hunk 2]
```diff
@@ -4428,5 +4428,15 @@ def check_unlimited():
         check_unlimited()
 
 
+def test_bsr_construct(self):
```
Comment (perimosocordiae) [inline scipy/sparse/tests/test_base.py]: This should be in the `TestBSR` class, maybe as a part of `test_constructor3`.

**Comment by ev-br** [comment]
Updated.

**Comment by perimosocordiae** [comment]
LGTM, merged.

**Pull request merged**